src/app/TideIndexApplication.cpp

```diff
@@ -1,1183 +1,1207 @@
+			  carp(CARP_FATAL, "Something went wrong whilst sorting, because sorted peptide file and peptide files have different sizes");
+		  }
+	  }else {
+		  carp(CARP_FATAL, "System sort failed to create the sorted petide file");
+	  }
+  } else {
+    sort(peptide_list.begin(), peptide_list.end(), less<TideIndexPeptide>());    
+  }
+ 
+  // Prepare the protocol buffer for the peptides.  
+  carp(CARP_INFO, "Writting peptides");
+
+  // pb::Header header_with_mods;
+  pb::Header_PeptidesHeader& pep_header = *(header_with_mods.mutable_peptides_header());
+  
+  pep_header.Clear();
+  pep_header.set_min_mass(min_mass);
+  pep_header.set_max_mass(max_mass);
+  pep_header.set_min_length(min_length);
+  pep_header.set_max_length(max_length);
+  pep_header.set_monoisotopic_precursor(monoisotopic_precursor);
+  pep_header.set_enzyme(enzyme);
+  if (enzyme != "no-enzyme") {
+    pep_header.set_full_digestion(digestion == FULL_DIGEST);
+    pep_header.set_max_missed_cleavages(missed_cleavages);
+  }
+  pep_header.mutable_mods()->CopyFrom(*(var_mod_table.ParsedModTable()));
+  pep_header.mutable_nterm_mods()->CopyFrom(*(var_mod_table.ParsedNtpepModTable()));
+  pep_header.mutable_cterm_mods()->CopyFrom(*(var_mod_table.ParsedCtpepModTable()));
+  pep_header.mutable_nprotterm_mods()->CopyFrom(*(var_mod_table.ParsedNtproModTable()));
+  pep_header.mutable_cprotterm_mods()->CopyFrom(*(var_mod_table.ParsedCtproModTable()));
+
+  pep_header.set_decoys_per_target(numDecoys);
+
+  header_with_mods.set_file_type(pb::Header::PEPTIDES);
+  header_with_mods.set_command_line(cmd_line);
+  pb::Header_Source* source = header_with_mods.add_source();
+  source->mutable_header()->CopyFrom(proteinPbHeader);
+  source->set_filename(AbsPath(out_proteins));
+
+  pb::Header header_no_mods;
+  header_no_mods.CopyFrom(header_with_mods);
+  pb::ModTable* del = header_no_mods.mutable_peptides_header()->mutable_mods();
+  del->mutable_variable_mod()->Clear();
+  del->mutable_unique_deltas()->Clear();
+
+  bool need_mods = var_mod_table.Unique_delta_size() > 0;
+
+  string peptidePbFile = need_mods ? modless_peptides : peakless_peptides;  
+  
+  // Check header
+  if (header_no_mods.source_size() != 1) {
+    carp(CARP_FATAL, "header_no_mods had a number of sources other than 1");
+  }
+  
+  headerSource = header_no_mods.mutable_source(0);
+  if (!headerSource->has_filename() || headerSource->has_filetype()) {
+    carp(CARP_FATAL, "pbHeader source invalid");
+  }
+
+  // Now check other desired settings
+  if (!header_no_mods.has_peptides_header()) {
+    carp(CARP_FATAL, "!header_no_mods->has_peptideHeapheader()");
+  }
+  const pb::Header_PeptidesHeader& settings = header_no_mods.peptides_header();
+  //if (!Peptide::SetMinMaxMassAndLength(settings)) {
+  //  carp(CARP_FATAL, "Error setting min/max mass/length");
+  if (!settings.has_enzyme() || settings.enzyme().empty()) {
+    carp(CARP_FATAL, "Enzyme settings error");

```

Comment:
This is not very informative. What should the user do to fix this problem?

Reply:
Hello @wsnoble , I believe this code was already there before we touched it.  If you could give me some context or suggestions I would make the changes to the text.

Reply:
I think in practice this if statement should never fire.  But a better error message would be something like `No digestion enzyme specified. Please check your settings for the "enzyme" and "custom-enzyme" parameters.`